src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala

```diff
@@ -195,7 +201,14 @@ object MissingLinkPlugin extends AutoPlugin {
       .filter(f => f.module.fold(true)(excluded))
       .map(_.artifact)
 
-    val projectArtifact = toArtifact(classDirectory)
+    val projectArtifact =
+      if (scanDependencies)
+        new ArtifactBuilder()
+          .name(new ArtifactName("project"))
+          .classes(runtimeArtifactsAfterExclusions.flatMap(_.classes.asScala).toMap.asJava)
+          .build()
```

Comment:
Please extract this is a separate method, near `private def toArtifact`.

Reply:
I have extracted a method here, as well as added two tests to cover both sides of this conditional. Please let me know if there's anything else needed to get this merged 🙂